crates/core/tedge_api/src/entity_store.rs

```diff
@@ -197,13 +198,28 @@ impl EntityStore {
         self.entities.get(entity_topic_id)
     }
 
+    /// Returns a mutable reference to the `EntityMetadata` for the given `EntityTopicId`.
+    pub fn get_mut(&mut self, entity_topic_id: &EntityTopicId) -> Option<&mut EntityMetadata> {
+        self.entities.get_mut(entity_topic_id)
+    }
```

Comment:
issue: Having this method be `pub` is unsafe, it allows the caller to set parent to an entity that doesn't exist, invalid `EntityExternalId`, etc. It also doesn't handle other affected entities: i.e. returning information that when a new entity is created, it affects its parent. This was part of the original specification, but AFAIK it's rarely used, so it's not a problem, but unchecked access to `EntityMetadata` is a problem.

This function is only used once in this module though, so we can just remove the `pub`.